pkg/runtime/k0s/init.go

```diff
@@ -36,46 +40,102 @@ func (k *Runtime) init() error {
 	return nil
 }
 
-// MergeConfigOnMaster0 convert the cluster file spec to kubectl.yaml (k0sctl config file) to lead cluster run
-func (k *Runtime) MergeConfigOnMaster0() error {
-	if err := k.K0sConfig.ConvertTok0sConfig(k.cluster); err != nil {
-		return fmt.Errorf("failed to convert to k0s config from clusterfile: %v", err)
+// GenerateConfigOnMaster0 generate the k0s.yaml to /etc/k0s/k0s.yaml and lead the controller node install
+func (k *Runtime) GenerateConfigOnMaster0() error {
+	if err := k.generateK0sConfig(); err != nil {
+		return fmt.Errorf("failed to generate config: %v", err)
 	}
-	ssh, err := k.getHostSSHClient(k.cluster.GetMaster0IP())
-	if err != nil {
-		return fmt.Errorf("failed to get ssh client: %v", err)
+	if err := k.modifyConfigRepo(); err != nil {
+		return fmt.Errorf("failed to modify config to private repository: %s", err)
+	}
+	return nil
+}
+
+// GenerateCert generate the containerd CA for registry TLS and k0s join token for 100 years.
+func (k *Runtime) GenerateCert() error {
+	if err := k.generateK0sToken(); err != nil {
+		return err
+	}
+
+	if err := k.GenerateRegistryCert(); err != nil {
+		return err
 	}
-	output, err := ssh.Cmd(k.cluster.GetMaster0IP(), GetK0sVersionCMD)
+	return k.SendRegistryCert(k.cluster.GetMasterIPList()[:1])
+}
+
+func (k *Runtime) generateK0sConfig() error {
+	master0IP := k.cluster.GetMaster0IP()
+	ssh, err := k.getHostSSHClient(master0IP)
 	if err != nil {
 		return err
 	}
 
-	k.K0sConfig.DefineConfigFork0s(string(output), k.RegConfig.Domain, k.RegConfig.Port, k.cluster.Name)
+	mkdirCMD := "mkdir -p /etc/k0s"
+	if _, err := ssh.Cmd(master0IP, mkdirCMD); err != nil {
+		return err
+	}
 
-	//write k0sctl.yaml to master0 rootfs
-	if err := k.K0sConfig.WriteConfigToMaster0(k.getRootfs()); err != nil {
+	configCreateCMD := fmt.Sprintf("k0s config create > %s", DefaultK0sConfigPath)
+	if _, err := ssh.Cmd(master0IP, configCreateCMD); err != nil {
+		return err
+	}
+	if err != nil {
```

Comment:
duplicated `if err != nil`, you can remove this line.

Reply:
I had forget to remove this line since last change😕 , my fault